recipe/cmake-tests/CMakeLists.txt

```diff
@@ -0,0 +1,44 @@
+cmake_minimum_required(VERSION 3.17)
+
+project(diana)
+
+option(CUDA_FINDER "Which method find_package will use: CUDA or CUDAToolkit" "CUDA")
+option(WITH_ENABLE_LANGUAGE "Use enable_language(CUDA)" OFF)
+message(STATUS "WITH_ENABLE_LANGUAGE=${WITH_ENABLE_LANGUAGE} CUDA_FINDER=${CUDA_FINDER}")
+
+set(VALID_CUDA_FINDERS "CUDA" "CUDAToolkit")
+if(NOT ${CUDA_FINDER} IN_LIST VALID_CUDA_FINDERS)
+    message(FATAL_ERROR "CUDA_FINDER must be one of: ${VALID_CUDA_FINDERS}")
+endif()
+
+file(GLOB cpu_source_files "${CMAKE_SOURCE_DIR}/src/*.cc")
+file(GLOB gpu_source_files "${CMAKE_SOURCE_DIR}/src/*.cu")
+file(GLOB include_files "${CMAKE_SOURCE_DIR}/src/*.hpp")
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
+
+if(WITH_ENABLE_LANGUAGE)
+    enable_language(CUDA)
+    set(CMAKE_CUDA_STANDARD 14)
+    set(CMAKE_CUDA_STANDARD_REQUIRED TRUE)
+    find_package("${CUDA_FINDER}" REQUIRED)
+    add_executable(diana ${cpu_source_files} ${gpu_source_files})
+
+else(WITH_ENABLE_LANGUAGE)
+    add_executable(diana ${cpu_source_files})
+
+    if("${CUDA_FINDER}" STREQUAL "CUDA")
+        message(STATUS "With CUDA")
+        find_package("${CUDA_FINDER}" REQUIRED)
+        set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -Xcompiler ,\"-std=c++14\"")
+        CUDA_ADD_LIBRARY(diana_gpu ${gpu_source_files})
+    else("${CUDA_FINDER}" STREQUAL "CUDA")
+        find_package("${CUDA_FINDER}" REQUIRED)
+        add_library(diana_gpu ${gpu_source_files})
```

Comment:
If `enable_language(CUDA)` isn't called it's not valid to try to build `.cu` files. It would be valid to build CPU source files `.cc` that use CUDA headers / libraries, but we'd need to add a `target_link_libraries` for example.

Alternatively, I think the main thing we want to test here is that the targets / variables created by `FindCUDAToolkit` / `FindCUDA` are populated with values as expected, so we don't necessarily need to compile anything.

Reply:
Instead of doing `add_library` here I think it would be good enough to check that `CUDAToolkit_LIBRARY_ROOT` (requires CMake 3.18+) is set as expected for `FindCUDAToolkit`.

SImilarly, we be good to check that `CUDA_TOOLKIT_ROOT_DIR` is set as expected for `FindCUDA` above in addition to doing the `CUDA_ADD_LIBRARY`.

Reply:
Added in last commit.